lib/generators/delivery/delivery-func.ts

```diff
@@ -0,0 +1,204 @@
+    console.log(`Generating '${chalk.yellow('delivery')}' models\n`);
+
+    const { contentTypeFiles, snippetFiles, taxonomyFiles, moduleResolution, environmentInfo, systemFiles } = await getFilesAsync(config);
+
+    await createFilesAsync(
+        {
+            contentTypeFiles,
+            snippetFiles,
+            taxonomyFiles,
+            environmentInfo,
+            systemFiles
+        },
+        {
+            formatOptions: config.formatOptions,
+            moduleResolution: moduleResolution,
+            outputDir: config.outputDir,
+            addTimestamp: config.addTimestamp
+        }
+    );
+
+    console.log(chalk.green(`\nCompleted`));
+}
+
+async function getFilesAsync(config: GenerateDeliveryModelsConfig): Promise<{
+    readonly contentTypeFiles: readonly GeneratedFile[];
+    readonly snippetFiles: readonly GeneratedFile[];
+    readonly taxonomyFiles: readonly GeneratedFile[];
+    readonly systemFiles: readonly GeneratedFile[];
+    readonly moduleResolution: ModuleResolution;
+    readonly environmentInfo: Readonly<EnvironmentModels.EnvironmentInformationModel>;
+}> {
+    const moduleResolution: ModuleResolution = getDefaultModuleResolution(config.moduleResolution);
+    const kontentFetcher = _kontentFetcher({
+        environmentId: config.environmentId,
+        apiKey: config.apiKey,
+        baseUrl: config.baseUrl
+    });
+
+    const environmentInfo = await kontentFetcher.getEnvironmentInfoAsync();
+
+    const [taxonomies, types, snippets, languages, collections, workflows] = await Promise.all([
+        kontentFetcher.getTaxonomiesAsync(),
+        kontentFetcher.getTypesAsync(),
+        kontentFetcher.getSnippetsAsync(),
+        kontentFetcher.getLanguagesAsync(),
+        kontentFetcher.getCollectionsAsync(),
+        kontentFetcher.getWorkflowsAsync()
+    ]);
+
+    const deliveryGenerator = deliveryContentTypeGenerator({
+        moduleResolution: moduleResolution,
+        environmentData: {
+            environment: environmentInfo,
+            types,
+            snippets,
+            taxonomies,
+            languages,
+            collections,
+            workflows
+        },
+        fileResolvers: config.fileResolvers,
+        nameResolvers: config.nameResolvers
+    });
+
+    const { contentTypeFiles, snippetFiles } = deliveryGenerator.generateModels();
+
+    const taxonomyFiles = deliveryTaxonomyGenerator({
+        moduleResolution: moduleResolution,
+        environmentData: {
+            environment: environmentInfo,
+            taxonomies: taxonomies
+        },
+        fileResolvers: config.fileResolvers,
+        nameResolvers: config.nameResolvers
+    }).generateTaxonomyTypes();
+
+    return {
+        contentTypeFiles,
+        snippetFiles,
+        taxonomyFiles,
+        moduleResolution,
+        environmentInfo,
+        systemFiles: deliveryGenerator.getSystemFiles()
+    };
+}
+
+async function createFilesAsync(
```

Comment:
The `createFilesAsync` and the exported function is almost the same in all the `...-func.ts` files. I believe it can be pretty easily abstracted.

Reply:
I knew you would point this out :D I was not too fond of this either, but it requires a better of representing sets of files which I initially didn't want to spend much time on. However, since you mentioned it, I refactored the way files are created removed all the `createFilesAsync` from the `*func` files and used the file manager directly.